magento-research/pwa-studio
# feat: remove esm mid-compilation and add babel preset

## Description

- Add a package defining [Babel preset](https://github.com/jamiebuilds/babel-handbook/blob/master/translations/en/user-handbook.md#making-your-own-preset) for all PWA Studio frontend code to use. (#1403) Move all Babel config into that preset and use the preset in project packages.
- Remove intermediate `esm` compilations, which are no longer necessary.
- Remove all compilation from `peregrine` package, which is only for use in compiled apps anyway and should lazily recompile as a dependency.
- Since Peregrine no longer has a direct compilation target, rename its `src` directory to `lib`, per the rules laid out in #1403.

## Related Issue
Closes #1403.

## Verification Steps

1. Clone the repository.
2. Run an install and a full build.
3. Observe that it's faster!
4. Check unit test results for preset package.
## Proposed Labels for Change Type/Package




This is totally for 4.0.0.


## Checklist:
